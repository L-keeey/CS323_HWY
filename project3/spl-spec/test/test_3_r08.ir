Error type 2 at Line 10: function should be invoked with a definition.
Error type 7 at Line 10: unmantching operands,such as adding an integer to a structure variable
FUNCTION mod :
PARAM v1
PARAM v2
t2 := v1
t6 := v1
t7 := v2
t4 := t6 / t7
t5 := v2
t3 := t4 * t5
t1 := t2 - t3
RETURN t1
FUNCTION DigitSum :
PARAM v3
t8 := v3
t9 := #0
IF t8 == t9 GOTO label1
GOTO label2
LABEL label1 :
t10 := #0
RETURN t10
LABEL label2 :
t14 := v3
t15 := #10
ARG t15
ARG t14
t12 := CALL mod
t17 := v3
t18 := #10
t16 := t17 / t18
ARG t16
t13 := CALL DigitSum
t11 := t12 + t13
RETURN t11
FUNCTION main :
READ t20
v4 := t20
t19 := v4
t21 := v4
t22 := #0
IF t21 < t22 GOTO label3
GOTO label4
LABEL label3 :
t24 := #1
t23 := #0 - t24
WRITE t23
GOTO label5
LABEL label4 :
t26 := v4
ARG t26
t25 := CALL DigitSum
WRITE t25
LABEL label5 :
t27 := #0
RETURN t27
