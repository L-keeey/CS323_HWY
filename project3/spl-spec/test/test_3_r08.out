Error type 2 at Line 10: function should be invoked with a definition.
Error type 7 at Line 10: unmantching operands,such as adding an integer to a structure variable
Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: mod
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: x
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: n
        RP
      CompSt (2)
        LC
        StmtList (3)
          Stmt (3)
            RETURN
            Exp (3)
              Exp (3)
                ID: x
              MINUS
              Exp (3)
                Exp (3)
                  LP
                  Exp (3)
                    Exp (3)
                      ID: x
                    DIV
                    Exp (3)
                      ID: n
                  RP
                MUL
                Exp (3)
                  ID: n
            SEMI
        RC
    ExtDefList (6)
      ExtDef (6)
        Specifier (6)
          TYPE: int
        FunDec (6)
          ID: DigitSum
          LP
          VarList (6)
            ParamDec (6)
              Specifier (6)
                TYPE: int
              VarDec (6)
                ID: y
          RP
        CompSt (7)
          LC
          StmtList (8)
            Stmt (8)
              IF
              LP
              Exp (8)
                Exp (8)
                  ID: y
                EQ
                Exp (8)
                  INT: 0
              RP
              Stmt (9)
                RETURN
                Exp (9)
                  INT: 0
                SEMI
            StmtList (10)
              Stmt (10)
                RETURN
                Exp (10)
                  Exp (10)
                    ID: mod
                    LP
                    Args (10)
                      Exp (10)
                        ID: y
                      COMMA
                      Args (10)
                        Exp (10)
                          INT: 10
                    RP
                  PLUS
                  Exp (10)
                    ID: DigitSum
                    LP
                    Args (10)
                      Exp (10)
                        Exp (10)
                          ID: y
                        DIV
                        Exp (10)
                          INT: 10
                    RP
                SEMI
          RC
      ExtDefList (13)
        ExtDef (13)
          Specifier (13)
            TYPE: int
          FunDec (13)
            ID: main
            LP
            RP
          CompSt (14)
            LC
            DefList (15)
              Def (15)
                Specifier (15)
                  TYPE: int
                DecList (15)
                  Dec (15)
                    VarDec (15)
                      ID: num
                SEMI
            StmtList (16)
              Stmt (16)
                Exp (16)
                  Exp (16)
                    ID: num
                  ASSIGN
                  Exp (16)
                    ID: read
                    LP
                    RP
                SEMI
              StmtList (17)
                Stmt (17)
                  IF
                  LP
                  Exp (17)
                    Exp (17)
                      ID: num
                    LT
                    Exp (17)
                      INT: 0
                  RP
                  Stmt (18)
                    Exp (18)
                      ID: write
                      LP
                      Args (18)
                        Exp (18)
                          MINUS
                          Exp (18)
                            INT: 1
                      RP
                    SEMI
                  ELSE
                  Stmt (20)
                    Exp (20)
                      ID: write
                      LP
                      Args (20)
                        Exp (20)
                          ID: DigitSum
                          LP
                          Args (20)
                            Exp (20)
                              ID: num
                          RP
                      RP
                    SEMI
                StmtList (21)
                  Stmt (21)
                    RETURN
                    Exp (21)
                      INT: 0
                    SEMI
            RC
Program
ExtDefList
ExtDef
FunDec
FUNCTION mod :
VarList
ParamDec
VarDec
PARAM v1
VarList
ParamDec
VarDec
PARAM v2
ComSt
DefList
StmtList
Stmt
Exp
Exp
t2 := v1
Exp
Exp
Exp
Exp
t6 := v1
Exp
t7 := v2
t4 := t6 / t7
Exp
t5 := v2
t3 := t4 * t5
t1 := t2 - t3
RETURN t1
StmtList
ExtDefList
ExtDef
FunDec
FUNCTION DigitSum :
VarList
ParamDec
VarDec
PARAM v3
ComSt
DefList
StmtList
Stmt
cond_Exp
3
Exp
t8 := v3
Exp
const
INT
0
t9 := #0
IF t8 == t9 GOTO label1
GOTO label2
LABEL label1 :
Stmt
Exp
const
INT
0
t10 := #0
RETURN t10
LABEL label2 :
StmtList
Stmt
Exp
Exp
Args
Exp
t14 := v3
Args
Exp
const
INT
0
t15 := #10
Args
